src/games/strategy/triplea/ai/proAI/util/ProMoveOptionsUtils.java

```diff
@@ -1163,8 +1175,77 @@ private void findBombardOptions(final PlayerID player, final List<Territory> myU
+        isEnemyCapitalOrFactory = true;
+      }
+      if (patd.getMaxBattleResult().getWinPercentage() < MIN_WIN_PERCENTAGE && isEnemyCapitalOrFactory
+          && alliedAttackOptions.getMax(t) != null) {
+
+        // Check for allied attack
+        final ProTerritory alliedAttack = alliedAttackOptions.getMax(t);
+        final Set<Unit> alliedUnits = new HashSet<Unit>(alliedAttack.getMaxUnits());
+        alliedUnits.addAll(alliedAttack.getMaxAmphibUnits());
+        if (!alliedUnits.isEmpty()) {
+
+          // Get max enemy defenders
+          final Set<Unit> additionalEnemyDefenders = new HashSet<Unit>();
+          final PlayerID alliedPlayer = alliedUnits.iterator().next().getOwner();
+          final List<PlayerID> players = utils.getOtherPlayersInTurnOrder(player);
+          for (final ProTerritory enemyDefendOption : enemyDefendOptions.getAll(t)) {
+            final Set<Unit> enemyUnits = new HashSet<Unit>(enemyDefendOption.getMaxUnits());
+            enemyUnits.addAll(enemyDefendOption.getMaxAmphibUnits());
+            if (!enemyUnits.isEmpty()) {
+              final PlayerID enemyPlayer = enemyUnits.iterator().next().getOwner();
+              if (utils.isPlayersTurnFirst(players, enemyPlayer, alliedPlayer)) {
+                additionalEnemyDefenders.addAll(enemyUnits);
+              }
+            }
+          }
+
+          // Check allied result without strafe
+          final Set<Unit> enemyDefendersBeforeStrafe = new HashSet<Unit>(defenders);
+          enemyDefendersBeforeStrafe.addAll(additionalEnemyDefenders);
+          final ProBattleResult result =
+              battleUtils.estimateAttackBattleResults(alliedPlayer, t, new ArrayList<Unit>(alliedUnits),
+                  new ArrayList<Unit>(enemyDefendersBeforeStrafe), alliedAttack.getMaxBombardUnits());
+          if (result.getWinPercentage() < WIN_PERCENTAGE) {
+            patd.setStrafing(true);
+
+            // Try to strafe to allow allies to conquer territory
+            final Set<Unit> combinedUnits = new HashSet<Unit>(patd.getMaxUnits());
+            combinedUnits.addAll(patd.getMaxAmphibUnits());
+            final ProBattleResult strafeResult =
+                battleUtils.callBattleCalculator(player, t, new ArrayList<Unit>(combinedUnits), defenders,
+                    patd.getMaxBombardUnits(), true, true);
+
+            // Check allied result with strafe
+            final Set<Unit> enemyDefendersAfterStrafe = new HashSet<Unit>(strafeResult.getAverageDefendersRemaining());
+            enemyDefendersAfterStrafe.addAll(additionalEnemyDefenders);
+            patd.setMaxBattleResult(battleUtils.estimateAttackBattleResults(alliedPlayer, t, new ArrayList<Unit>(
+                alliedUnits), new ArrayList<Unit>(enemyDefendersAfterStrafe), alliedAttack.getMaxBombardUnits()));
+
+
+            ProLogger.debug("Checking strafing territory: " + t + ", alliedPlayer="
```

Comment:
So I actually want the second part of what you are describing. When I simulate a game with all AI's I need to have output that shows essentially input data into strafing decision and output on whether it chose to strafe. That is what they few debugs I have now are trying to capture. Until I build out something like you describe then I need to put a few debug's into these decisions to print out info about what's going on.